### PR TITLE
Update to Membrane Core 0.11, modify storage cleanup functionality

### PR DESCRIPTION
**Pull request by balins**
- update to Core 0.11
- replace `{:cleanup, cleanup_fun}` notification with `cleanup_after` option, so cleanup runs after specified timeout if `Sink`/`SinkBin` was configured to do so
- bump package version to `0.10.0`

### CODE REVIEW

File: lib/membrane_http_adaptive_stream/sink_bin.ex [hunk 1]
```diff
@@ -189,49 +186,41 @@ defmodule Membrane.HTTPAdaptiveStream.SinkBin do
...
+            child({:payloader, ref}, payloader),
+            child({:cmaf_muxer, ref}, muxer),
+            bin_input(pad)
+            |> get_child({:payloader, ref})
+            |> get_child({:cmaf_muxer, ref}),
+            get_child(:audio_tee)
+            |> get_child({:cmaf_muxer, ref}),
+            get_child({:cmaf_muxer, ref})
+            |> via_in(pad, options: track_options(context))
+            |> get_child(:sink)
```
Comment (mat-hek) [inline lib/membrane_http_adaptive_stream/sink_bin.ex:211]: Let's make it more concise

File: lib/membrane_http_adaptive_stream/sink_bin.ex [hunk 2]
```diff
@@ -151,27 +168,7 @@ defmodule Membrane.HTTPAdaptiveStream.SinkBin do
...
-        nil -> nil
-        duration -> duration.target
-      end
-
-    [
-      track_name: track_name,
-      segment_duration: segment_duration,
-      target_partial_segment_duration: target_partial_segment_duration
-    ]
+    {[spec: {structure, []}], state}
```
Comment (mat-hek) [inline lib/membrane_http_adaptive_stream/sink_bin.ex:171]:
```suggestion
    {[spec: structure], state}
```

File: lib/membrane_http_adaptive_stream/sink_bin.ex [hunk 3]
```diff
@@ -247,75 +236,87 @@ defmodule Membrane.HTTPAdaptiveStream.SinkBin do
       |> Map.update!(:streams_to_start, update_streams_count)
       |> Map.update!(:streams_to_end, update_streams_count)
 
-    {{:ok, spec: spec}, state}
+    {[spec: {structure, []}], state}
```
Comment (mat-hek) [inline lib/membrane_http_adaptive_stream/sink_bin.ex:239]:
```suggestion
    {[spec: structure], state}
```

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: lib/membrane_http_adaptive_stream/sink.ex [hunk 4]
```diff
@@ -483,11 +493,39 @@ defmodule Membrane.HTTPAdaptiveStream.Sink do
...
+         manifest: manifest,
+         storage: storage,
+         cleanup_after: cleanup_after
+       }) do
+    {:ok, _pid} =
+      Task.start(fn ->
+        to_remove = Manifest.all_segments_per_track(manifest)
+        timeout = Membrane.Time.as_milliseconds(cleanup_after)
+
+        Process.sleep(timeout)
```
[the remark this answers is not in the document]
Comment (balins) [reply lib/membrane_http_adaptive_stream/sink.ex:517]: This is what we do in line 515

[the remark this answers is not in the document]
Comment (mat-hek) [reply lib/membrane_http_adaptive_stream/sink.ex:517]: no idea why I haven't noticed 🤷

**Review by mat-hek — APPROVED** [review]

**Pull request merged**